TestBangazonAPI/TestProductTypes.cs

```diff
@@ -0,0 +1,149 @@
+
+namespace TestBangazonAPI
+{
+    public class TestProductTypes
+    {
+        [Fact]
+        public async Task Test_Get_All_ProductTypes()
+        {
+            using (var client = new APIClientProvider().Client)
+            {
+                var response = await client.GetAsync("/producttype");
+
+                response.EnsureSuccessStatusCode();
+
+                string responseBody = await response.Content.ReadAsStringAsync();
+                var productTypes = JsonConvert.DeserializeObject<List<ProductType>>(responseBody);
+
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                Assert.True(productTypes.Count > 0);
+            }
+        }
+        [Fact]
+        public async Task Test_Get_Single_ProductType()
+        {
+
+            using (var client = new APIClientProvider().Client)
+            {
+                var response = await client.GetAsync("/producttype/2");
+
+                response.EnsureSuccessStatusCode();
+
+                string responseBody = await response.Content.ReadAsStringAsync();
+                var productType = JsonConvert.DeserializeObject<ProductType>(responseBody);
+
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                Assert.Equal("Stuffed Toy", productType.Name);
+                Assert.NotNull(productType);
+            }
+        }
+
+        [Fact]
+        public async Task Test_Get_NonExistant_ProductType_Fails()
+        {
+
+            using (var client = new APIClientProvider().Client)
+            {
+                var response = await client.GetAsync("/producttype/999999999");
+                Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+            }
+        }
+
+
+        [Fact]
+        public async Task Test_Create_And_Delete_ProductType()
+        {
+            using (var client = new APIClientProvider().Client)
+            {
+                ProductType toys = new ProductType
+                {
+                    Name = "Toys"
+                };
+                var toysAsJSON = JsonConvert.SerializeObject(toys);
+
+
+                var response = await client.PostAsync(
+                    "/ProductType",
+                    new StringContent(toysAsJSON, Encoding.UTF8, "application/json")
+                );
+
+                response.EnsureSuccessStatusCode();
+
+                string responseBody = await response.Content.ReadAsStringAsync();
+                var newProductType = JsonConvert.DeserializeObject<ProductType>(responseBody);
+
+                Assert.Equal(HttpStatusCode.Created, response.StatusCode);
+                Assert.Equal("Toys", newProductType.Name);
+
+
+                var deleteResponse = await client.DeleteAsync($"/productType/{newProductType.Id}");
+                deleteResponse.EnsureSuccessStatusCode();
+                Assert.Equal(HttpStatusCode.NoContent, deleteResponse.StatusCode);
+            }
+        }
+
+        [Fact]
```

Comment:
This test is in here twice! And this one wouldn't work anyway.

Reply:
Not sure what you're referring to.  More specifics?

Reply:
I'm not reading closely enough. Ignore this.